### PR TITLE
Update documentation in `telestion-api`

### PR DESCRIPTION
**Pull request by fussel178**
### Summary 

Add package information, add missing class and method documentation and add GitHub usernames to authors.

### Details 

Migrated content from #440 

### Additional information 

n/a

### CLA

- [x] I have signed the individual contributor's license agreement **and sent** it to the board of the WüSpace e. V. organization.

### CODE REVIEW

**Comment by fussel178** [comment]
@cb0s Please review this PR, make changes and mark it as "Ready for review" if you happy with it. :wink:

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/package-info.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Contains helper classes for the message handling for the Vert.x event bus system.<br/>
+ * This package adds support for Java-15 Records.
+ * <h3>Structure</h3>
+ * In the base package exist classes handling the general structure for messages which are sent via the event bus of
+ * Vert.x. JSON (Jackson Codec) is used to encode and decode messages for the event bus.
+ * <p>
+ * {@link de.wuespace.telestion.api.message.HeaderInformation} add support for message headers. They can be
+ * used to include data unrelated to the data itself of a message.
+ * <p>
+ * It is heavily recommended to use {@link de.wuespace.telestion.api.verticle.trait.WithEventBus}, it contains different
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/package-info.java:13]:
```suggestion
 * It is heavily recommended to use Telestion traits like {@link de.wuespace.telestion.api.verticle.trait.WithEventBus}, it contains different
```
Might good to say that WithEventBus is just an example.

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/package-info.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
+ * <h2>Telestion API - Telestion Verticle</h2>
+ *
+ * This package contains different classes helping with Vert.x verticles by providing often used methods which help
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/package-info.java:4]:
```suggestion
 * This package contains different classes helping with Vert.x verticles by providing commonly used methods which help
```

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/package-info.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * to improve the developer experience by massively reducing boiler-plate code.
+ * <h3>Usage</h3>
+ * Verticles within a Telestion Project should extend {@link de.wuespace.telestion.api.verticle.TelestionVerticle}.
+ * It provides different core functionality like more clean start() and stop() methods.
+ * <p>
+ * A Verticle usually needs a Configuration class. They must be created manually - the convention is to create a Record
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/package-info.java:10]:
```suggestion
 * A Verticle usually needs an implementation of {@link de.wuespace.telestion.api.verticle.TelestionConfiguration}. They must be created manually - the convention is to create a Record
```

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/package-info.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * in the same class called Configuration. By inferring the generics of the
+ * {@link de.wuespace.telestion.api.verticle.TelestionVerticle} with this Configuration, the loading code for
+ * configurations gets reduced massively.<br/>
+ * When using no Configuration it is recommended to use {@link de.wuespace.telestion.api.verticle.GenericConfiguration}
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/package-info.java:14]:
```suggestion
 * {@link de.wuespace.telestion.api.verticle.TelestionVerticle} with this {@code Configuration}, the loading code for
 * configurations gets reduced massively.<br/>
 * When using no {@code Configuration} it is recommended to use {@link de.wuespace.telestion.api.verticle.GenericConfiguration}
```

File: modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/trait/package-info.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+/**
+ * <h2>Telestion API - Telestion Verticle Traits</h2>
+ *
+ * This package contains different traits to enhance the functionality of
+ * {@link de.wuespace.telestion.api.verticle.TelestionVerticle} even further.
+ * <p>
+ * Traits contain default methods, which are "short-cuts" for the most-used Vert.X features in the Telestion Ecosystem.
```
Comment (cb0s) [inline modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/trait/package-info.java:7]:
```suggestion
 * Traits contain default methods, which are "short-cuts" for the most-used Vert.x features in the Telestion Ecosystem.
```

**Review by cb0s — CHANGES_REQUESTED** [review]
Thanks @fussel178 for adding our previous work. After reading through it again, I stumbled upon some minor things I would improve.

However, I think it is ready for review for the others as well :D

**Review by cb0s — APPROVED** [review]
Perfection 👍 ✔️

**Pull request merged**